src/tests/ftest/io/nvme_object.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+import threading
+import avocado
+
+from daos_api import DaosApiError
+from test_utils import TestPool, TestContainer
+from apricot import TestWithServers
+
+
+def container_write(container, record):
+    """Method to write to a container
+
+       Args:
+         container: instance of TestContainer
+         record: record size to be written
+    """
+    # update record qty
+    container.record_qty.update(record)
+    print("\nRecord Size:{}".format(container.record_qty))
+
+    # write multiple objects
+    container.write_objects()
+
+    # read written objects and verify
+    container.read_objects()
+
+def test_runner(self, size, record_size):
+    """Method to perform simultaneous writes of varying
+       record size to a container in a pool
+
+       Args:
+         self: avocado test object
+         size: pool size to be created
+         record_size (list): list of different record sizes to be written
+    """
+    # pool initialization
+    pool = TestPool(self.context, self.log)
+    pool.get_params(self)
+    # set pool size
+    pool.nvme_size.update(size)
+    # Create a pool
+    pool.create()
+    pool.connect()
+
+    # invoke pool query before write
+    pool.get_info()
+
+    # create container
+    container = TestContainer(pool)
+    container.get_params(self)
+    container.create()
+    container.open()
+
+    threads = []
+    for rec in record_size:
+        for thread in range(2):
```

Comment:
So now structure looks like below correct ? What happen if we increase the range 2 to 16?

- 1> Pool Size 1
- -       A> Record Size 1
- - -              X> thread range 1
- - -              Y> thread range 2
- -       B> Record Size 2
- -       C> Record Size 3
- -       D> Record Size 4
- 2> Pool Size 2
- 3> Pool size 3

Reply:
It should simply write 16 more processes and I think it should pass without issues.